storage/postgres/storage.go

```diff
@@ -112,9 +112,7 @@ func (ps *PostgresStorage) Close() error {
 	ps.checkOpen()
 	ps.mutex.Lock()
 	defer ps.mutex.Unlock()
-	defer func() {
-		ps.db = nil
-	}()
+
```

Comment:
With this change can you reopen the storage?

Reply:
different SMs dont share the same storage object anymore so different test suites can "reopen" a storage (it would esencially be a new storage).

But within the same SM process you cant reopen a storage and you shouldnt be able to, imo - opening should be done only one time